sync/sync.go

```diff
@@ -0,0 +1,97 @@
+package sync
+
+import (
+	"context"
+	"time"
+
+	"gorm.io/gorm"
+
+	"github.com/photon-storage/go-common/log"
+
+	"github.com/photon-storage/photon-explorer/chain"
+	"github.com/photon-storage/photon-explorer/database/orm"
+)
+
+// EventProcessor is the processor for synchronizing photon change events.
+type EventProcessor struct {
+	syncSeconds uint64
+	db          *gorm.DB
+	node        *chain.Node
+}
+
+// NewEventProcessor returns the new instance of EventProcessor.
+func NewEventProcessor(syncSeconds uint64, nodeURL string, db *gorm.DB) *EventProcessor {
+	return &EventProcessor{
+		syncSeconds: syncSeconds,
+		db:          db,
+		node:        chain.NewNode(nodeURL),
+	}
+}
+
+// Run executing the timing task of processing chain data.
+func (e *EventProcessor) Run(ctx context.Context) {
+	ticker := time.NewTicker(time.Duration(e.syncSeconds) * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			localSlot, localHash, err := e.localChainStatus()
```

Comment:
I dont think we need to query DB again and again. We know the chain status from the last update.